worker/src/direct_invocation/watch_list_service.rs

```diff
@@ -0,0 +1,63 @@
+/*
+    Copyright 2019 Supercomputing Systems AG
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+use dashmap::DashMap;
+use sp_core::H256 as Hash;
+use substratee_worker_primitives::RpcResponse;
+use ws::Sender;
+
+#[derive(Clone)]
+pub struct WatchingClient {
+    pub client: Sender,
+    pub response: RpcResponse,
+}
+
+/// Trait to manage the watched list of direct invocation requests
+pub trait WatchListFacade {
+    fn add_watching_client(&self, hash: Hash, client: WatchingClient);
+
+    fn remove_watching_client(&self, hash: &Hash);
+
+    fn get_watching_client(&self, hash: &Hash) -> Option<WatchingClient>;
+}
+
+pub struct WatchListService {
+    clients: DashMap<Hash, WatchingClient>,
+}
+
+impl WatchListService {
+    pub fn new() -> Self {
+        WatchListService {
+            clients: DashMap::<Hash, WatchingClient>::new(),
+        }
+    }
+}
+
+impl WatchListFacade for WatchListService {
```

Comment:
Why do you add the `Facade` suffix? I don't understand what information it should convey.

Reply:
yes agreed, I suppose I could get rid of it 👍